web/source/kmwbase.ts

```diff
@@ -710,7 +710,11 @@ namespace com.keyman {
 
       PKbd = PKbd || this.core.activeKeyboard;
 
-      return com.keyman.osk.VisualKeyboard.buildDocumentationKeyboard(PKbd, argFormFactor, argLayerId, this.osk.computedHeight);
+      // help.keyman.com will set this function in place, so we'll give it priority.
+      const getOskHeight = this['getOskHeight'];
+      let targetHeight = (typeof getOskHeight == 'function' ? getOskHeight() : this.osk.computedHeight) || 200;
```

Comment:
On further consideration, I'm still not sure this is right. The priority should be: `getOskHeight`, if available, then if it returns nullish, `this.osk.computedHeight`, and if still nullish, then `200`.

```suggestion
      let targetHeight = (typeof getOskHeight == 'function' ? getOskHeight() : null) ?? this.osk.computedHeight ?? 200;
```

I don't think that `getOskHeight` should ever return nullish but at least in ios-host.js, it could return `0` if `getOskHeight()` is ever called before `setOskHeight()`. Could certainly happen in a race condition.

So that's a good safety measure?

Reply:
Why is the proposed change necessary?

Code paths:
- getOskHeight exists
    - returns a valid number
        - that number is used
    - (somehow) returns `null` / `undefined`
        - `null || 200` resolves to `200`
        - `undefined || 200` resolves to `200`
- `getOskHeight` not defined
    - `this.osk.computedHeight` is now given preference instead.
        - it holds a valid number
            - we use that number
        - it returns `null` or `undefined`
            - `null || 200` and `undefined || 200` both resolve to `200`.

Is there something I'm missing in this breakdown?  Because, if not, the original already covers things perfectly.

Reply:
By the way, this reminds me of why I'd selected the original code version before applying https://github.com/keymanapp/keyman/pull/6793#discussion_r899814867's changes:  if `getOskHeight()` were to return a nullish value, `computedHeight` could take over.  Not so after its changes, though.

So... which kind of fallback behavior do we want?  Should I revert my 'accept' on that prior change?

Reply:
... ah, right.  Took me a bit to work through, but I see it now.  (`computedHeight` as a first-priority fallback.)